### ISSUE TITLE
Failing to cross-compile when using gonative, gox and Go 1.5.1

### ISSUE THREAD

**Issue by tchap** [inconshreveable/gonative#29]
I am using `gonative` and `gox` to cross-compile an executable using `os/user`. It has always been working, but it broke down when CircleCI upgraded their Golang to 1.5.1 I think. Now I am getting `user: Current not implemented on darwin/amd64` from the cross-compiled executables. Any ideas what might be wrong now before I try to dive deep into Go 1.5 cross-compilation changes myself?

CC @inconshreveable

**Comment by tchap** [comment]
Hmm, works locally, I will close this before I can prove it's not just CircleCI env setup error...

**Comment by inconshreveable** [comment]
It works locally on 1.5.1 but not on circle? That's pretty bizarre.

**Comment by tchap** [comment]
Yeah, something weird. When I do everything locally and try the cross-compiled executable, it works. When I SSH into a CircleCI container and do the same thing manually, I get the error that `user.Current` is not specified...

**Comment by inconshreveable** [comment]
Some possibilities:

- Inside circle, the binaries are being built with their version of Go instead of the one created by gonative
- gonative is failing silently and producing a Go installation that doesn't used the downloaded libraries

**Comment by tchap** [comment]
Thanks for the hints. What is weird for me is that it has always been working just fine, it just broke for some reason recently. I did refactor the build scripts, but they should be doing the same thing...

**Comment by tchap** [comment]
Well, the good part is that I changed the scripts to work locally as well and the executables are broken even when run locally, so the scripts are simply wrong somehow. No idea why it used to work then, but whatever...

**Comment by tchap** [comment]
Actually I am getting the same error locally when using the commands as in README examples, i.e.

```bash
$ go get -u github.com/mitchellh/gox
$ <build current gonative using make>
$ cd /your/project
$ gonative build
$ PATH=$PWD/go/bin/:$PATH gox
```

Weird, I though that this was working...

**Comment by tchap** [comment]
Actually it doesn't work even when I use system Go 1.5.2 like

```
gox -osarch=darwin/amd64
```

It starts working when I add `-cgo`...

**Comment by tchap** [comment]
Seems like a `gox` issue to me, somehow...

**Comment by tchap** [comment]
Not an expert on cross-compiling, but this is strange...

```bash
$ GOOS=darwin GOARCH=amd64 GOROOT="`pwd`/go" CGO_ENABLED=0 go build -v
...
os/user
...
```

I was expecting Go to re-use `os/user` as downloaded by `gonative`, not to re-compile it again... And of course it this scenario I am getting `user.Current` not implemented error.

**Comment by tchap** [comment]
Using `CGO_ENABLED=1` works of course, but that is exactly what I cannot use when cross-compiling...

**Comment by tchap** [comment]
I basically don't understand why running `PATH="`pwd`/go/bin:$PATH" GOOS=linux GOARCH=amd64 CGO_ENABLED=1 go build -v` rebuilds the standard library...

```bash
 PATH="`pwd`/go/bin:$PATH" GOOS=linux GOARCH=amd64 CGO_ENABLED=1 go build -v
runtime
errors
math
sync/atomic
unicode/utf8
sort
unicode
sync
...
```

**Comment by tchap** [comment]
The command actually rebuilds the stdlib even with `CGO_ENABLED=0`...

**Comment by tchap** [comment]
Yeah, confirmed that the same thing totally works when using `--version 1.4.3` with `gonative`. Forwarded to golang-nuts mailing list...

**Comment by databus23** [comment]
I can't seem to get genitive to work for go >= 1.5. It always seems to rebuild the stdlib.

**Comment by databus23** [comment]
Maybe its related to this change: https://go-review.googlesource.com/#/c/9154/.

**Comment by tchap** [comment]
Dunno. I actually asked on the mailing list, but no one replied. I ended up downgrading to Go 1.4.3 actually...
